### PR TITLE
Create UI settings tab and move UI related settings there

### PR DESCRIPTION
**Pull request by adipose**

### CODE REVIEW

**Comment by adipose** [comment]
image

**Comment by adipose** [comment]
This could probably use some testing, as code to control these options is moving from 3 different source files to a new one.

**Comment by adipose** [comment]
https://mega.nz/file/JFREGKIR#j8AVYIJORJ0F8QQzgwGTc5wgDWnfWWtdYhdoD3tvo0Y

Test build.

**Comment by xLn2** [comment]
1. Seekbar

There is some issues about initializing active passive buttons on seek bar
Enable Hover > Apply > close settings page and open settings again.
Disable Hover > Apply > close settings page and open settings again.

image

image

image

Bug: While `hover type` is Video Preview, Enabling or disabling `Hover` doesn't function. Still shows video preview.

- Suggestion 1: Disabling Hover would effect on ComboBox2, Static4, ComboBox3, Static5 and Edit3 (Gather these info from Window spy software)
image

2. Modern theme

Suggestion 1: Disabling Modern theme should disable Modern Theme Color and modern seekbar height (Static1, ComboBox1, Static3 and Edit2) because disabling modern theme it uses legacy theme I suppose.
image

Suggestion 2: Since there is no theme except modern ones I think "_Modern_ Theme Color" without Modern is more suitable.

3. Renaming suggestions

Suggestion 1: `Hide Windowed Controls` should has better name. I suggest "Auto-hide controls ~~and panels~~ in windowed _mode_"
Suggestion 2:  `Default Toolbar Size` -> "Default size of controls"

**Comment by clsid2** [comment]
> Suggestion 1: Hide Windowed Controls should has better name. I suggest "Hide controls and panels in windowed mode"
> Suggestion 2: Default Toolbar Size -> "Default size of controls"

1 That is too long and translations are typically even longer.
2 Maybe "Control buttons size"

**Comment by adipose** [comment]
Easy fixes probably.  I will be away for a few days.

Also noticed two spellings of Seekbar.

**Comment by adipose** [comment]
For modern seekbar, we may want to make it the default?

If classic seekbar will still be kept around, via advanced, we'd still need to disable the height for consistency.

@clsid2 ?

**Comment by clsid2** [comment]
Use modern seekbar for modern theme only.

**Comment by xLn2** [comment]
1. Preview width % of screen editbox (Edit3) shouldn't allow non digit character. Preventing at the beginning is better I think. All other options allow digit only.
image

2. Options have static width so if it is not make any trouble -bugs, lag etc-, expanding option list's width with its largest member (like `add to favorites`'s combobox) might be useful. (especially for seek bar options)

3. Since there modern theme is new normal for a time, Modern theme color and modern seek bar height might be simplificated as `Theme color` and `Seekbar height`

>  2 Maybe "Control buttons size"

This is better.


**Thank you for your great efforts.🎉**

**Comment by adipose** [comment]
> * Options have static width so if it is not make any trouble -bugs, lag etc-, expanding option list's width with its largest member (like `add to favorites`'s combobox) might be useful. (especially for seek bar options)

There is code to auto expand in mpc-hc.  Is it needed for translations?

> 
> * Since there modern theme is new normal for a time, Modern theme color and modern seek bar height might be simplificated as `Theme color` and `Seekbar height`

Only if we get rid of classic mode, otherwise it is confusing to classic mode users.

**Comment by xLn2** [comment]
> There is code to auto expand in mpc-hc. Is it needed for translations?

Yes. For some languages last few characters don't seen. Example, 'seekbar' is translating  based as 'progress bar' in some languages (same situation with progress bar to localized string) and this lead to different string length.

**Comment by adipose** [comment]
> > There is code to auto expand in mpc-hc. Is it needed for translations?
> 
> Yes. For some languages last few characters don't seen. Example, 'seekbar' is translating based as 'progress bar' in some languages (same situation with progress bar to localized string) and this lead to different string length.

OK, that's working.  Worst one is

`Per sobre de la barra de navegació`

This fits now.  Another idea is maybe just to save "above" or "below" as it's already the seekbar section.  But this is clearer.

**Comment by adipose** [comment]
@clsid2 , any other changes needed here?

File: src/mpc-hc/PPageAdvanced.cpp [hunk 1]
```diff
@@ -136,14 +133,7 @@ void CPPageAdvanced::InitSettings()
         m_list.SetItemData(iItem, nItem);
     };
 
-    addBoolItem(HIDE_WINDOWED, IDS_RS_HIDE_WINDOWED_CONTROLS, false, s.bHideWindowedControls, StrRes(IDS_PPAGEADVANCED_HIDE_WINDOWED));
```
Comment (clsid2) [inline src/mpc-hc/PPageAdvanced.cpp:139]: HIDE_WINDOWED and other removed options can be removed from the enum.

Then it can be merged I think.

**Comment by adipose** [comment]
I renamed one thing, and it's going to force a bunch of retranslations:

```
 msgctxt "IDS_TIME_TOOLTIP_BELOW"
-msgid "Below seek bar"
-msgstr "搜尋列之下"
+msgid "Below seekbar"
+msgstr ""

```

I'm not sure if Transifex allows for me to just rename the original string and keep the translations.

**Comment by xLn2** [comment]
These strings also currently has translations. If there is way to use these again might be useful.
```
Show OSD (requires restart)
Use enhanced taskbar features
Snap to desktop edges
Show chapter marks on seek bar
Language
Limit window proportions on resize
```

Language:  ->  Languages:     _(maybe)_

Also 
Options > Tweaks > Fast seek (on keyframe) checkbox doesn't show tooltip.

**Comment by adipose** [comment]
> Options > Tweaks > Fast seek (on keyframe) checkbox doesn't show tooltip.

Fixed.

To preserve old translations, we could add them to the string table.

**Comment by tsubasanouta** [comment]
About ”Seek Bar” on View Menu
Would you rename "See&k Bar" to "See&kbar" ?

**Comment by clsid2** [comment]
With and without space is both correct imo, so pointless change that breaks translations (unless manually editing po).

**Pull request merged**